amplipi/streams.py

```diff
@@ -803,12 +805,24 @@ def info(self) -> models.SourceInfo:
           line = line.strip()
           if line:
             data = line.split(',,,')
+            if self.track != data[1]: # When song changes, stop inverting state
+              self.invert_liked_state = False
             source.state = self.state
             source.artist = data[0]
             source.track = data[1]
+            self.track = data[1]
             source.album = data[2]
             source.img_url = data[3].replace('http:', 'https:') # HACK: kind of a hack to just replace with https
-            source.rating = models.PandoraRating(int(data[4]))
+
+            # Pianobar doesn't update metadata after a song starts playing
+            # so when you like a song you have to change the state manually until next song
+            if self.invert_liked_state and int(data[4]) == 0:
+                source.rating = models.PandoraRating(1)
+            elif self.invert_liked_state and int(data[4]) == 1:
+              source.rating = models.PandoraRating(0)
+            else:
+              source.rating = models.PandoraRating(int(data[4]))
+
```

Comment:
```suggestion
            initial_rating = models.PandoraRating(int(data[4]))
            source.rating = initial_rating
            if self.invert_liked_state:
              if initial_rating == models.PandoraRating.DEFAULT:
                source.rating = models.PandoraRating.LIKED
              elif initial_rating == models.PandoraRating.LIKED:
                source.rating = models.PandoraRating.DEFAULT
```

Reply:
I think this reads a tad better since it uses the actual Pandora rating types instead of magic/unnamed integers.